client/src/store/modules/products.js

```diff
@@ -0,0 +1,35 @@
+import axios from '@/plugins/axios';
+
+const initialState = {
+    products: []
+};
+
+const state = () => initialState;
+
+const getters = {
+    getProducts: state => state.products
+};
+
+const mutations = {
+    setProducts: (state, products) => {
+        state.products = products.sort((a, b) => a.name < b.name);
```

Comment:
I think here you are working with reference

Reply:
so should it be `[...products].sort((a, b) => a.name < b.name);`?

Reply:
`sort` also should returns `-1`, `1` and `0` and I think there should be `-` instead of `<` 
and about reference, cant you pass just already sorted array?

Reply:
done